### ISSUE TITLE
Verify Chennai area districts

### ISSUE THREAD

**Issue by planemad** [justinelliotmeyers/India_Official_Boundaries_2019#9]
From https://github.com/justinelliotmeyers/India_Official_Boundaries_2019/issues/4#issuecomment-608710635 the following districts in Tamil Nadu are still to be updated correctly due to some subdistrict level reorganisation and a lack of updated reference maps:
- Chennai
- Thiruvallur
- Chengalpattu
- Kanchipuram

It might be possible to use the [village boundaries](https://github.com/justinelliotmeyers/official_india_2011_village_boundary_polygons) and the [village lists from LGD](https://github.com/planemad/india-local-government-directory) to piece together the updated shape.

**Comment by justinelliotmeyers** [comment]
@planemad have you seen any official maps of the new boundaries? I had to reference some maps of the pre-adjusted boundaries to create the current divisions. 

Taluks/ Talukas are different than subdivisions correct?

Any detailed data or info going into this will help me figure out how to adjust the output dataset. Want to keep @aadityadar looped in on this as well.

Thanks!

**Comment by aadityadar** [comment]
will let you know if i find anything. fixing sub-districts is a herculean task! 
ps- i think you are right about taluka being different from sub-divisions.

**Comment by justinelliotmeyers** [comment]
@aadityadar @planemad is this correct? https://twitter.com/chennaicorp/status/1251915534746058752/photo/1

**Comment by planemad** [comment]
This is the shape of Chennai city according to the official source
<img width="778" alt="Screen Shot 2020-04-19 at 9 05 20 PM" src="https://user-images.githubusercontent.com/126868/79704892-8ad79e00-8281-11ea-9f4c-18cdaa041c2a.png"><br>
https://twitter.com/chennaicorp/status/1251914948738863104/photo/1

The only shapefile i know of is at datameet which seems to match, except for zone XI: 
<img width="239" alt="Screen Shot 2020-04-19 at 9 07 14 PM" src="https://user-images.githubusercontent.com/126868/79704973-e3a73680-8281-11ea-8aee-17655d19a4e2.png"><br>https://github.com/datameet/Municipal_Spatial_Data/blob/master/Chennai/Wards.geojson

**Comment by planemad** [comment]
This map on the GCC website seems to match the datameet shape though.

<img width="488" alt="Screen Shot 2020-04-19 at 9 13 02 PM" src="https://user-images.githubusercontent.com/126868/79705130-924b7700-8282-11ea-8761-d833ee001ae6.png">
<br>http://chennaicorporation.gov.in/images/MAP_ZONE_WARD.jpg

**Comment by justinelliotmeyers** [comment]
@planemad ive been uploading all maps I'm finding and sources. I think the difficult thing is Chennai is a city that has been growing and officials want the city to expand further. Its almost in a situation like Hyderabad - so much expansion, proposed expansion, and nobody has kept track of it. Id like to get something showing an actual date on the map if possible. Can we somehow email them or have someone there send us a current official map?

**Comment by planemad** [comment]
> so much expansion, proposed expansion, and nobody has kept track of it

Theres no confusion here. The legal description of the extent of each of the 200 wards in the new city limits (which was done in 2018) is in this Government notification (in Tamil) http://www.chennaicorporation.gov.in/images/ward_details.pdf . Theres been no change in city limits since then.

The problem is to find digital vectors for the same.

**Comment by planemad** [comment]
FWIW, this is the list of villages listed under Chennai District in the [LGD](https://github.com/planemad/india-local-government-directory/blob/master/village-directory.csv). I'm not sure if the subdistricts listed here are correct, they seem like the old subdistricts before they were absorbed into the expanded Chennai district. Also the old city limit is listed in 1 row without a name, so not sure how much to trust the LGD for Chennai.

<details>

State Code | State Name (In English) | Census 2011 Code | District Code | District Name (In English) | Census 2011 Code | Sub district Code | Subdistrict Name (In English) | Census 2011 Code_1 | Village Code | Village Name (In English) | Census 2011 Code_2 | Local Body Code | Local Body Name (In English) | Census 2011 Code_3 | Local Body Type Code | Local Body Type Name | Block Code | Block Name (In English)
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5701 | Mathavaram | 05701 | 933875 | Chettimedu | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5701 | Mathavaram | 05701 | 933859 | Madhavaram | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5701 | Mathavaram | 05701 | 933874 | Kosapur | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5701 | Mathavaram | 05701 | 933872 | Mathur | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5701 | Mathavaram | 05701 | 933873 | Manchambakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5701 | Mathavaram | 05701 | 933860 | Puzhal | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5701 | Mathavaram | 05701 | 933857 | Puthagaram | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5704 | Alandur | 05704 | 933821 | Adambakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5704 | Alandur | 05704 | 933806 | Madanandapuram | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5704 | Alandur | 05704 | 933818 | Alandur | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5704 | Alandur | 05704 | 933830 | Meenambakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5704 | Alandur | 05704 | 933814 | Manappakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5704 | Alandur | 05704 | 933816 | Nandambakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5704 | Alandur | 05704 | 933813 | Mugalivakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5701 | Mathavaram | 05701 | 933862 | Vadaperumbakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5701 | Mathavaram | 05701 | 933855 | Surapet | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5701 | Mathavaram | 05701 | 933871 | Villlakupattu | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5701 | Mathavaram | 05701 | 933864 | Villakupattu | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5705 | Sholinganallur | 05705 | 933849 | Jaladampet | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5705 | Sholinganallur | 05705 | 933866 | Enjambakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5705 | Sholinganallur | 05705 | 933854 | Kottivakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5705 | Sholinganallur | 05705 | 933867 | Karapakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5705 | Sholinganallur | 05705 | 933863 | Neelangarai | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5705 | Sholinganallur | 05705 | 933848 | Madipakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5705 | Sholinganallur | 05705 | 933856 | Palavakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5705 | Sholinganallur | 05705 | 933865 | Okkiam Thorapakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5705 | Sholinganallur | 05705 | 933853 | Perungudi | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5705 | Sholinganallur | 05705 | 933850 | Pallikaranai | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5705 | Sholinganallur | 05705 | 933870 | Semmanjeri | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5705 | Sholinganallur | 05705 | 933861 | Seevaram | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5705 | Sholinganallur | 05705 | 933844 | Ullagaram | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5705 | Sholinganallur | 05705 | 933868 | Sholinganallur | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5982 | Tiruvottiyur | 00000 | 933852 | Chinnasekkadu | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5705 | Sholinganallur | 05705 | 933869 | Uthandi | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5982 | Tiruvottiyur | 00000 | 933846 | Ernavoor | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5982 | Tiruvottiyur | 00000 | 933842 | Elandanchery | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5982 | Tiruvottiyur | 00000 | 933845 | Kathivakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5982 | Tiruvottiyur | 00000 | 629181 | Kadapakkam | 629181 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5982 | Tiruvottiyur | 00000 | 933843 | Sadayankuppam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5982 | Tiruvottiyur | 00000 | 933841 | Manali | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5982 | Tiruvottiyur | 00000 | 933847 | Thiruvottiyur | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5982 | Tiruvottiyur | 00000 | 933851 | Sathangadu | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 |   |   |   |   |   |   |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5704 | Alandur | 05704 | 933825 | Pazhavanthangal | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5704 | Alandur | 05704 | 933827 | Nanganallur | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5700 | Ambattur | 05700 | 933879 | Ambattur | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5704 | Alandur | 05704 | 933828 | Thalakanancheri | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5700 | Ambattur | 05700 | 933883 | Kakkapallam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5700 | Ambattur | 05700 | 933880 | Attipattu | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5700 | Ambattur | 05700 | 933881 | Mannurpet | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5700 | Ambattur | 05700 | 933882 | Korattur | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5700 | Ambattur | 05700 | 933884 | Mogapair | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5700 | Ambattur | 05700 | 933877 | Menambedu | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5700 | Ambattur | 05700 | 933885 | Paadi | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5700 | Ambattur | 05700 | 933876 | Oragadam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5981 | Maduravoyal | 00000 | 933839 | Chettiyar Agaram | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5700 | Ambattur | 05700 | 933878 | Patravakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5981 | Maduravoyal | 00000 | 933831 | Maduravoyal | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5981 | Maduravoyal | 00000 | 933835 | Karampakkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5981 | Maduravoyal | 00000 | 933834 | Nolambur | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5981 | Maduravoyal | 00000 | 933836 | Nerkundram | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5981 | Maduravoyal | 00000 | 933837 | Ramapuram | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5981 | Maduravoyal | 00000 | 933833 | Porur | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5981 | Maduravoyal | 00000 | 933840 | Thandalam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5981 | Maduravoyal | 00000 | 933838 | Sivabootham | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5701 | Mathavaram | 05701 | 629180 | Ariyalur | 629180 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5981 | Maduravoyal | 00000 | 933832 | Valasaravalkam | 000000 |   |   |   |   |   |   |  
33 | TAMIL NADU | 33 | 568 | CHENNAI | 603 | 5701 | Mathavaram | 05701 | 933858 | Kathirvedu | 000000 |   |   |   |   |   |  

</details>

**Comment by justinelliotmeyers** [comment]
http://chennaicorporation.gov.in/delimitation_draft/citymap_majorroads.pdf has 2018 mentioned. I'll keep reviewing.

**Comment by justinelliotmeyers** [comment]
I rectified the Tamil version against the datameet map. I will continue to look for another day or two before I plan on a partial revert of Chennai and update of city/ district boundary. I would really like to get an updated/ official map from them if possible. I know these things aren't easily accessible - but Chennai is a 'smart city' so... maybe this is possible. I tagged an issue in datameet and hopefully @devdattaT can help. Really appreciate all of your help @planemad and @aadityadar 

![c_align](https://user-images.githubusercontent.com/5324316/79711010-b021d780-8294-11ea-8456-139d6e76eeb4.jpg)

**Comment by justinelliotmeyers** [comment]
emailed https://play.google.com/store/apps/details?id=com.gcc.smartcity&hl=en_US

**Comment by aadityadar** [comment]
> emailed https://play.google.com/store/apps/details?id=com.gcc.smartcity&hl=en_US

you are very optimistic :-)

**Comment by planemad** [comment]
> emailed https://play.google.com/store/apps/details?id=com.gcc.smartcity&hl=en_US

From this thumbnail, the Chennai boundary looks like the old one 🙈 

<img width="199" alt="Screen Shot 2020-04-20 at 3 59 29 PM" src="https://user-images.githubusercontent.com/126868/79794288-01c37400-8320-11ea-9c0f-5156d93e01ed.png">

**Comment by justinelliotmeyers** [comment]
I was told to contact: https://www.annauniv.edu/RemoteSensing/

**Comment by sharun-s** [comment]
Not sure if this is better but the stopcorona.tn.gov.in has an arcgis dashboard
Check out the services-
https://services9.arcgis.com/HwXIp55hAoiv6DE9/ArcGIS/rest/services

There is a Tamil_Nadu_Boundaries_WFL1 service that returns district shapes

**Comment by justinelliotmeyers** [comment]
@sharun-s pulling in now! stay tuned, i will keep you posted. Thank you!!!!!!

**Comment by justinelliotmeyers** [comment]
@planemad @aadityadar new data sources uploaded. I am thinking of implementing the Chennai data, but need to take this new source into account. Just wanted to ping both of you for input.  Thanks!

**Comment by justinelliotmeyers** [comment]
@sharun-s the Chennai boundary somehow follows the data I previously built... I am not sure how. Could they have used data from this repo to update official government maps??

**Comment by aadityadar** [comment]
i would have been very surprised if the data on the stopcorona website was using the latest boundaries; in the govt, one arm rarely knows what the other is doing

**Comment by sharun-s** [comment]
@justinelliotmeyers looks like TN is way behind the curve compared to other states gis offerings :( 
(I found that server and this thread while searching for Chennai city's zone wise boundaries) 
Anyways thanks for checking it out and maintaining such a useful repo!

**Comment by justinelliotmeyers** [comment]
Updating to Chennai proper city limits.

![tn_chennai_update_04292020](https://user-images.githubusercontent.com/5324316/80661801-b74ba100-8a5d-11ea-8a9e-24b163611f28.jpg)

**Comment by justinelliotmeyers** [comment]
Inserted city proper boundaries. Thanks for everyone's help and time with this!

**Comment by planemad** [comment]
Adding some more official documentation regarding the reorganisation of Chennai district from this 2018 notification on https://lgdirectory.gov.in/globalviewsubdistrict.do

**tl;dr**
- Chennai dsitrict is coterminus with Greater Chennai Corporation boundary
- No change in total subdistricts, subdsitricts overlapping expanded Chennai subdsitrict retained, the subdistrict part outside Chennai are absorbed into neighboring subdistricts
- Finally Chennai district should have 16 subdistricts (10 existing+6 due to expansion), but the lgd site still shows only the 6 due to expansion. The district site has the full list, so LGD subdistrict list is not as reliable as one would hope.

**References**

<img width="467" alt="Screen Shot 2020-07-24 at 11 04 47 AM" src="https://user-images.githubusercontent.com/126868/88406081-26efd480-cd9e-11ea-816a-487972174add.png">
<img width="602" alt="Screen Shot 2020-07-24 at 11 01 29 AM" src="https://user-images.githubusercontent.com/126868/88405392-30c50800-cd9d-11ea-8ef5-83140e0de52e.png">
<img width="590" alt="Screen Shot 2020-07-24 at 11 01 38 AM" src="https://user-images.githubusercontent.com/126868/88405394-31f63500-cd9d-11ea-8c58-d36d30ba04a2.png">

<img width="886" alt="Screen Shot 2020-07-24 at 11 03 57 AM" src="https://user-images.githubusercontent.com/126868/88406079-26efd480-cd9e-11ea-9395-04ba9c57eab8.png">
<img width="996" alt="Screen Shot 2020-07-24 at 11 21 02 AM" src="https://user-images.githubusercontent.com/126868/88407211-c82b5a80-cd9f-11ea-9b63-2988590e4682.png">
https://chennai.nic.in/about-district/administrative-setup/revenue-administration/#tlks

**Comment by justinelliotmeyers** [comment]
@planemad just started poking around again for Chennai... and I found something. **updated** was a random old service with outdated boundaries... :(

**Comment by leclercprof** [comment]
Is there anybody who have a geojson file of pre-2011 ward map for chennai ?
